_posts/2015-12-07-storm-compatibility.md

```diff
@@ -0,0 +1,155 @@
+categories: news
+---
+
+[Apache Storm](https://storm.apache.org) was one of the first distributed and scalable stream processing systems available in the open source space offering (near) real-time tuple-by-tuple processing semantics.
+Initially released by the developers at Backtype in 2011 under the Eclipse open-source license, it became popular very quickly in the stream processor space.
+Only shortly afterwards, Twitter acquired Backtype.
+Since then, Storm has been growing in popularity and is used in production at big companies.
+In 2013, Storm entered the Apache incubator program, followed by its graduation to top-level in 2014.
+
+Apache Flink is a stream processing engine that improves upon older technologies like Storm in several dimensions,
+including [strong consistency guarantees](https://ci.apache.org/projects/flink/flink-docs-master/internals/stream_checkpointing.html) ("exactly once"),
+a higher level [DataStream API](https://ci.apache.org/projects/flink/flink-docs-master/apis/streaming_guide.html),
+support for [event time and a rich windowing system](http://flink.apache.org/news/2015/12/04/Introducing-windows.html),
+as well as [superior throughput with competitive low latency](https://data-artisans.com/high-throughput-low-latency-and-exactly-once-stream-processing-with-apache-flink/).
+
+While Flink offers several technical benefits over Storm, an existing investment on a codebase of applications developed over Storm often makes it difficult to switch engines.
+For these reasons, as part of the Flink 0.10 release, Flink ships with a Storm compatibility package that allows users to:
+
+* Run **unmodified** Storm topologies using Apache Flink benefiting from superior performance.
+* **Embed** Storm code (spouts and bolts) as operators inside Flink DataStream programs.
+
+Only minor code adaptations are required in order to submit the program to Flink instead of Storm.
+This minimizes the work for a developer to run an existing Storm topology while leveraging Apache Flink’s fast and robust execution engine.
+
+We note that the Storm compatibility package is continuously improving and does not cover the full spectrum of Storm’s API.
+However, it is powerful enough to cover many use cases.
+
+## Executing Storm topologies with Flink
+
+<center>
+<img src="{{site.baseurl}}/img/blog/flink-storm.png" style="height:200px;margin:15px">
+</center>
+
+The easiest way to use the Storm compatibility package is by executing a whole Storm topology in Flink.
+For this, you only need to include `flink-storm` to your Storm project and **change a two lines of code** in your original Storm program.
+Furthermore, you need to package an uber-jar (also called fat-jar) that contains your code as well as Flink and Storm dependencies.
+If you want to avoid huge jar files, you can also manually copy the compatibility- and Storm core-jar into Flink's lib folder before starting up the Flink cluster.
```

Comment:
Just too much information if you just want to casually read about the compatibility layer.